statistics/handle/lockstats/unlock_stats.go

```diff
@@ -26,20 +26,18 @@ import (
 
 const (
 	selectDeltaSQL = "SELECT count, modify_count, version FROM mysql.stats_table_locked WHERE table_id = %?"
-	updateDeltaSQL = "UPDATE mysql.stats_meta SET version = %?, count = count + %?, modify_count = modify_count + %? WHERE table_id = %?"
+	updateDeltaSQL = "UPDATE mysql.stats_meta SET version = %?, count = CASE WHEN %? < 0 THEN IF(count > ABS(%?), count - ABS(%?), 0) ELSE count + %? END, modify_count = modify_count + %? WHERE table_id = %?"
```

Comment:
@time-and-fate Do you have any ideas about how to improve this?

Reply:
Looks like these two expressions are equivalent.
```sql
CASE WHEN %? < 0 THEN
    IF(count > ABS(%?), count - ABS(%?), 0)
ELSE
    count + %?
END
```
```sql
IF(count + %? > 0, count + %?, 0)
```

Reply:
Good point!

Reply:
Updated in [`e852581` (#47319)](https://github.com/pingcap/tidb/pull/47319/commits/e8525818f7531a95fd3d7774345131148612cff8)